### ISSUE TITLE
STYLE003: Detect inconsistent SQL keyword case

### ISSUE THREAD

**Issue by RAprogramm** [RAprogramm/sql-query-analyzer#19]
## Summary

Add a rule to detect inconsistent casing of SQL keywords within a single query (e.g., mixing `SELECT` with `from` or `Where`).

## Why This Matters

Consistent keyword casing is a code quality issue:

- **Readability** — mixed case is visually jarring and harder to scan
- **Team standards** — inconsistency suggests lack of formatting discipline
- **Professional appearance** — code reviews flag this immediately
- **Maintenance** — consistent style makes patterns easier to recognize

### The Professional Standard

Most SQL style guides recommend one of:

1. **UPPERCASE keywords** (most common in enterprise)
   ```sql
   SELECT id, name FROM users WHERE status = 'active' ORDER BY name;
   ```

2. **lowercase keywords** (common in modern tooling)
   ```sql
   select id, name from users where status = 'active' order by name;
   ```

Both are valid — but mixing them is not:

```sql
-- This is ugly and unprofessional:
SELECT id, name from Users WHERE status = 'active' Order By name;
```

## Rule Specification

| Property | Value |
|----------|-------|
| Rule ID | `STYLE003` |
| Name | Inconsistent keyword case |
| Category | Style |
| Severity | Info |
| Default | Enabled |

### Detection Patterns

```sql
-- Should trigger (mixed case)
SELECT id FROM users where status = 'active';
select * From orders WHERE id = 1;
SELECT name, Select email FROM users;  -- even worse

-- Should NOT trigger (consistent)
SELECT * FROM users WHERE id = 1;
select * from users where id = 1;
```

### Keywords to Check

Primary: `SELECT`, `FROM`, `WHERE`, `JOIN`, `LEFT`, `RIGHT`, `INNER`, `OUTER`, `ON`, `AND`, `OR`, `NOT`, `IN`, `ORDER`, `BY`, `GROUP`, `HAVING`, `LIMIT`, `OFFSET`, `INSERT`, `INTO`, `VALUES`, `UPDATE`, `SET`, `DELETE`, `CREATE`, `ALTER`, `DROP`, `TABLE`, `INDEX`, `AS`, `DISTINCT`, `UNION`, `ALL`, `BETWEEN`, `LIKE`, `IS`, `NULL`, `EXISTS`, `CASE`, `WHEN`, `THEN`, `ELSE`, `END`

### Suggested Message

> "Inconsistent SQL keyword casing: found both 'SELECT' and 'from'. Use consistent case for all keywords (either UPPERCASE or lowercase) for better readability."

## References

- [SQL Style Guide by Simon Holywell](https://www.sqlstyle.guide/)
- [GitLab SQL Style Guide](https://about.gitlab.com/handbook/business-technology/data-team/platform/sql-style-guide/)
- [Mozilla SQL Style Guide](https://docs.telemetry.mozilla.org/concepts/sql_style.html)
- [Kickstarter SQL Style Guide](https://gist.github.com/fredbenenson/7bb92718e19138c20591)

## Implementation Notes

- Extract all SQL keywords from query
- Check if all are uppercase, all lowercase, or mixed
- Consider: PascalCase identifiers (table/column names) are fine
- Mixed case = trigger rule
- Don't flag: quoted identifiers, string literals
- Configurable: preferred style (UPPER vs lower) as optional enforcement

**Comment by RAprogramm** [comment]
Blocked on architecture: rules receive statements re-serialized from the sqlparser AST (verified: 'select id From users' reaches rules as 'SELECT id FROM users'), so the original keyword case is already normalized away. Implementing STYLE003 requires threading the pre-parse source text (or statement spans) into Query, which is a breaking change to the public Query struct. Deferring until a major/minor API revision carries that field.
